app/models/split_detail.rb

```diff
@@ -16,6 +16,12 @@ def initialize(params)
     super
   end
 
+  def variant_details
+    @variant_details ||= variants.map do |variant|
+      VariantDetail.find_or_initialize_by(split: split, variant: variant)
+    end
+  end
```

Comment:
i think this code is also nearly identical to the code we added to the SplitPresenter:

```ruby
class SplitPresenter < SimpleDelegator
  def variant_details
    variants.map do |variant|
      detail = VariantDetail.find_or_initialize_by(split: __getobj__, variant: variant)
      VariantPresenter.new(detail)
    end
  end
end
```

should we redefine the presenter in terms of this?